lib/eventsource/index.js

```diff
@@ -0,0 +1,448 @@
+  #withCredentials = false
+
+  #readyState = CONNECTING
+
+  #request = null
+  #controller = null
+
+  /**
+   * @type {object}
+   * @property {string} lastEventId
+   * @property {number} reconnectionTime
+   * @property {any} reconnectionTimer
+   */
+  #settings = null
+
+  /**
+   * Creates a new EventSource object.
+   * @param {string} url
+   * @param {EventSourceInit} [eventSourceInitDict]
+   * @see https://html.spec.whatwg.org/multipage/server-sent-events.html#the-eventsource-interface
+   */
+  constructor (url, eventSourceInitDict = {}) {
+    // 1. Let ev be a new EventSource object.
+    super()
+
+    webidl.argumentLengthCheck(arguments, 1, { header: 'EventSource constructor' })
+
+    if (!experimentalWarned) {
+      experimentalWarned = true
+      process.emitWarning('EventSource is experimental, expect them to change at any time.', {
+        code: 'UNDICI-ES'
+      })
+    }
+
+    url = webidl.converters.USVString(url)
+    eventSourceInitDict = webidl.converters.EventSourceInitDict(eventSourceInitDict)
+
+    // 2. Let settings be ev's relevant settings object.
+    this.#settings = {
+      lastEventId: '',
+      reconnectionTime: defaultReconnectionTime
+    }
+
+    // 1. Let baseURL be this's relevant settings object's API base URL.
+    const baseURL = getGlobalOrigin()
+
+    // 2. Let urlRecord be the result of applying the URL parser to url with baseURL.
+    let urlRecord
+
+    try {
+      // 3. Let urlRecord be the result of encoding-parsing a URL given url, relative to settings.
+      urlRecord = new URL(url, baseURL)
+    } catch (e) {
+      // 4. If urlRecord is failure, then throw a "SyntaxError" DOMException.
+      throw new DOMException(e, 'SyntaxError')
+    }
+
+    // 5. Set ev's url to urlRecord.
+    this.#url = urlRecord.href
+
+    // 6. Let corsAttributeState be Anonymous.
+    let corsAttributeState = ANONYMOUS
+
+    // 7. If the value of eventSourceInitDict's withCredentials member is true,
+    // then set corsAttributeState to Use Credentials and set ev's
+    // withCredentials attribute to true.
+    if (eventSourceInitDict.withCredentials) {
+      corsAttributeState = USE_CREDENTIALS
+      this.#withCredentials = true
+    }
+
+    // 8. Let request be the result of creating a potential-CORS request given
+    // urlRecord, the empty string, and corsAttributeState.
+    const initRequest = {
+      redirect: 'follow',
+      keepalive: true,
+      // @see https://html.spec.whatwg.org/multipage/urls-and-fetching.html#cors-settings-attributes
+      mode: 'cors',
+      credentials: corsAttributeState === 'anonymous'
+        ? 'same-origin'
+        : 'omit',
+      referrer: 'no-referrer',
+      referrerPolicy: 'no-referrer'
+    }
+
+    // 9. Set request's client to settings.
+
+    // 10. User agents may set (`Accept`, `text/event-stream`) in request's header list.
+    initRequest.headers = new HeadersList()
+    initRequest.headers.set('accept', 'text/event-stream', true)
```

Comment:
This isn't correct, the internal response has a `headersList` property. You should add this to initRequest:

```js
    const initRequest = {
      // ...
      headersList: [['accept', 'text/event-stream']] 
    }
```

Reply:
I will check it. The headers were actually really problematic.

Reply:
Please see my new remark below